cmd/kube-proxy/app/server.go

```diff
@@ -209,6 +209,8 @@ func (o *Options) AddFlags(fs *pflag.FlagSet) {
 
 	fs.Float32Var(&o.config.ClientConnection.QPS, "kube-api-qps", o.config.ClientConnection.QPS, "QPS to use while talking with kubernetes apiserver")
 	fs.Var(&o.config.DetectLocalMode, "detect-local-mode", "Mode to use to detect local traffic. This parameter is ignored if a config file is specified by --config.")
+	fs.StringVar(&o.config.InterfaceName, "interface-name", o.config.InterfaceName, "The interface name in the cluster. Kube-proxy considers traffic as local if originating from an interface which matches the value. This argument should be set if DetectLocalMode is set to LocalModeInterfaceName.")
+	fs.StringVar(&o.config.InterfacePrefix, "interface-prefix", o.config.InterfacePrefix, "The interface prefix in the cluster. Kube-proxy considers traffic as local if originating from interfaces that match the given prefix. This argument should be set if DetectLocalMode is set to LocalModeInterfacePrefix.")
```

Comment:
actually this is on me, the KEP: https://github.com/kubernetes/enhancements/blob/0e4d5df19d396511fe41ed0860b0ab9b96f46a2d/keps/sig-network/2450-Remove-knowledge-of-pod-cluster-CIDR-from-iptables-rules/README.md#design-details says `--pod-interface-prefix`.. Should I stick to that? I also like `--detect-local-traffic-interface-name` like you are suggesting.

Reply:
`--pod-interface-prefix` works for the InterfacePrefix case, but you can't really do `--pod-interface-name` for the InterfaceName case, because it's not the name of a pod interface, it's the name of the interface you go through on your way to the pod interface...

It could maybe be `--pod-bridge-interface-name` but then I might want to change the `--detect-local-mode` value to include "bridge" too?

Having a ridiculously long command line option name like `--detect-local-traffic-interface-name` is not necessarily bad because it encourages people to move away from command-line options and start using a YAML config file. :slightly_smiling_face:

Reply:
So let's go with `--pod-interface-prefix` and `--pod-bridge-interface-name` and I'll change it everywhere in this PR i.e replace interfaceprefix and interfacename with these two ^ ones. Will rebase with the changes in 24hrs... Just want to give some time for you both to ack these ^ fields.

Reply:
done.

Reply:
> done

forgot to `git push`?

Reply:
yeah done came out faster than I'd have liked. :D